cmd/EnqueueFFISDownload/handler.go

```diff
@@ -0,0 +1,164 @@
+	}
+
+	log.Info(logger, url, "Parsed URL from email body")
+
+	// Enqueue the URL for download
+	err = enqueueURLForDownload(url, sqsclient, ctx)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func plaintextMIMEFromEmailBody(email string) (string, error) {
+	reader := strings.NewReader(email)
+	msg, err := mail.ReadMessage(reader)
+	if err != nil {
+		return "", err
+	}
+	mediaType, params, err := mime.ParseMediaType(msg.Header.Get("Content-Type"))
+	if mediaType != "multipart/alternative" {
+		return "", fmt.Errorf("expected multipart/alternative, got %s", mediaType)
+	}
+	if err != nil {
+		return "", err
+	}
+	mr := multipart.NewReader(msg.Body, params["boundary"])
+	for {
+		p, err := mr.NextPart()
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			return "", err
+		}
+		if strings.HasPrefix(p.Header.Get("Content-Type"), "text/plain") {
+			buf := new(bytes.Buffer)
+			_, err := buf.ReadFrom(p)
+			if err != nil {
+				return "", err
+			}
+			return buf.String(), nil
+		}
+	}
+
+	return "", fmt.Errorf("no text/plain part found")
+}
+
+func parseURLFromEmailBody(email string) (string, error) {
+	plaintext, err := plaintextMIMEFromEmailBody(email)
+	if err != nil {
+		return "", err
+	}
+	patternRegex := regexp.MustCompile(urlPattern)
+	matches := patternRegex.FindStringSubmatch(plaintext)
+	if len(matches) == 0 {
+		return "", fmt.Errorf("no matches found")
+	}
+	if len(matches) > 1 {
+		return "", fmt.Errorf("multiple matches found")
+	}
+	return matches[0], nil
+}
+
+func getQueueUrl(queueName string, client SQSAPI, ctx context.Context) (*string, error) {
+	result, err := client.GetQueueUrl(ctx, &sqs.GetQueueUrlInput{
+		QueueName: aws.String(queueName),
+	},
+	)
+	if err != nil {
+		return nil, err
+	}
+	return result.QueueUrl, nil
+
+}
+
+func enqueueURLForDownload(url string, client SQSAPI, ctx context.Context) error {
+	queueUrl, err := getQueueUrl(destinationQueue, client, ctx)
+	if err != nil {
+		return err
+	}
+	log.Info(logger, *queueUrl, "Got SQS queue URL")
+
+	message := sqs.SendMessageInput{
+		MessageBody: aws.String(url),
+		QueueUrl:    aws.String(*queueUrl),
+	}
+	output, err := client.SendMessage(ctx, &message)
+
+	if err != nil {
+		return err
+	}
+	log.Info(logger, *output.MessageId, "Sent SQS message")
+	return nil
+}
+
+func getEmailFromS3Event(s3client S3API, s3Event events.S3Event, ctx context.Context) (string, error) {
+	bucket := s3Event.Records[0].S3.Bucket.Name
+	log.Debug(logger, bucket, "Reading from bucket")
+	uploadedFile := s3Event.Records[0].S3.Object.Key
+	log.Info(logger, uploadedFile, "New email file")
+	// Get the email body
+	resp, err := s3client.GetObject(ctx, &s3.GetObjectInput{
+		Bucket: aws.String(bucket),
+		Key:    aws.String(uploadedFile),
+	})
+	if err != nil {
+		return "", err
+	}
+	defer resp.Body.Close()
+	buf := new(bytes.Buffer)
+	_, err = buf.ReadFrom(resp.Body)
+	if err != nil {
+		return "", err
+	}
+	emailBody := buf.String()
+	return emailBody, nil
+}
```

Comment:
(Note: Commenting here but also pertains to `plaintextMIMEFromEmailBody` and potentially `parseURLFromEmailBody`.)

I think the handling of the content downloaded from S3 could be simplified a bit (potentially with some efficiency gains) by making greater use of the `io.Reader` provided by the S3 response (at `resp.Body`) instead of preemptively stringifying the file's contents. The main advantage in this approach is that it avoids holding the entire `.eml` file in memory, at most only holding onto the plaintext portion for the duration of the regex operation.

The following (untested) example code is meant to clarify the behavior, but not necessarily opine on how you ultimately choose to organize the code – the primary reason why the S3 operation is moved into `handleS3EventWithConfig` is to include closing the reader once the URL has been parsed, rather than for anything related to code-cleanliness and/or testability.
```go
func handleS3EventWithConfig(cfg aws.Config, ctx context.Context, s3Event events.S3Event, s3client S3API, sqsclient SQSAPI) error {
    // NB: Everything from here until defer resp.body.Close() may well be implemented in a separate function
    bucket := s3Event.Records[0].S3.Bucket.Name
	log.Debug(logger, bucket, "Reading from bucket")
	uploadedFile := s3Event.Records[0].S3.Object.Key
	log.Info(logger, uploadedFile, "New email file")

	// Get the email body
	resp, err := s3client.GetObject(ctx, &s3.GetObjectInput{
		Bucket: aws.String(bucket),
		Key:    aws.String(uploadedFile),
	})
	if err != nil {
		return nil, err
	}
	defer resp.Body.Close()

    ptBuf, err := plaintextMIMEFromEmailBody(resp.Body)
    if err != nil {
		return err
	}
    
    url, err := parseURLFromEmailBody(ptBuf)
    ... // cont.
}

func plaintextMIMEFromEmailBody(body io.Reader) (*bytes.Buffer, error) {
	msg, err := mail.ReadMessage(body)
	if err != nil {
		return nil, err
	}

	mediaType, params, err := mime.ParseMediaType(msg.Header.Get("Content-Type"))
	if mediaType != "multipart/alternative" {
		return nil, fmt.Errorf("expected multipart/alternative, got %s", mediaType)
	}
	if err != nil {
		return nil, err
	}
	mr := multipart.NewReader(msg.Body, params["boundary"])
	for {
		p, err := mr.NextPart()
		if err == io.EOF {
			break
		}
		if err != nil {
			return nil, err
		}
		if strings.HasPrefix(p.Header.Get("Content-Type"), "text/plain") {
			buf := new(bytes.Buffer)
			_, err := buf.ReadFrom(p)
			return buf, err
		}
	}
	return nil, fmt.Errorf("no text/plain part found")
}

func parseURLFromEmailBody(body *bytes.Buffer) (string, error) {
	patternRegex := regexp.MustCompile(urlPattern)	
	matches := patternRegex.FindSubmatch(body)
	if len(matches) == 0 {
		return "", fmt.Errorf("no matches found")
	}
	if len(matches) > 1 {
		return "", fmt.Errorf("multiple matches found")
	}
	return string(matches[0]), nil
}
```